audformat/core/utils.py

```diff
@@ -1034,6 +1055,11 @@ def map_file_path(
     else:
         index = index.map(func)
 
+    index = set_index_dtypes(
+        index,
+        {define.IndexField.FILE: 'string'},
+    )
```

Comment:
I think in the corresponding test `pd.testing.assert_index_equal` doe snot check for the correct dtype.
We need to use `pd.testing.assert_index_equal(..., exact=True)` or manualy test for the right dtype as the test should have failed before the addition of this code.

Reply:
I think it would it would fail if the `dtypes` do not match, e.g.:

```python
pd.testing.assert_index_equal(pd.Index(['a']), pd.Index(['a'], dtype='string'))
```
```
Attribute "dtype" are different
[left]:  object
[right]: string[python]
```

But I also tried to create an example where `string` is converted to `object` and couldn't find one. Unless you do e.g. `lambda x: 0`, but that does not match the type hint. So I guess we could also skip `set_index_dtypes()`. On the other hand, it has almost no overhead since it converts if the dtype is not `'string'`.

Reply:
I think it is indeed not possible to find a conversion function that converts to `object` from `string`.

When we leave the current code, we implicitly support mapping functions like `lambda x: 0` as everything that can be converted to a string will be converted to a string by it. So the question is if we want to have this or not?

Reply:
Ok, if you agree I will remove it.

Reply:
Yes, fine with me.

Reply:
done

Reply:
Looks like you forgot to push.